packages/ws/src/ws/WebSocketShard.ts

```diff
@@ -128,10 +128,28 @@ export class WebSocketShard extends AsyncEventEmitter<WebSocketShardEventsMap> {
 	}
 
 	public async connect() {
-		const promise = this.initialConnectResolved ? Promise.resolve() : once(this, WebSocketShardEvents.Ready);
+		const controller = new AbortController();
+		let promise;
+
+		if (!this.initialConnectResolved) {
+			// Sleep for the remaining time, but if the connection closes in the meantime, we shouldn't wait the remainder to avoid blocking the new conn
+			promise = Promise.race([
+				once(this, WebSocketShardEvents.Ready, { signal: controller.signal }),
+				once(this, WebSocketShardEvents.Resumed, { signal: controller.signal }),
+			]);
+
+			controller.abort();
```

Comment:
Won't this promise reject immediately, since you abort the controller as soon as you make the promise? 👀

Reply:
yea my bad, was moving stuff around and left that there